staging/src/k8s.io/client-go/tools/cache/controller_test.go

```diff
@@ -893,3 +897,230 @@ func (m *mockTxnStore) Transaction(txns ...Transaction) *TransactionError {
 	m.succeedCount = len(txns)
 	return nil
 }
+
+func TestReplaceEvents(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	source := fcache.NewFakeControllerSource()
+	store := NewStore(DeletionHandlingMetaNamespaceKeyFunc)
+	t.Cleanup(func() {
+		source.Shutdown()
+	})
+
+	recorder := newEventRecorder(store)
+
+	fifo := NewRealFIFOWithOptions(RealFIFOOptions{
+		KnownObjects: store,
+	})
+
+	cfg := &Config{
+		Queue:            fifo,
+		ListerWatcher:    source,
+		ObjectType:       &v1.Pod{},
+		FullResyncPeriod: 0,
+
+		Process: func(obj interface{}, isInInitialList bool) error {
+			if deltas, ok := obj.(Deltas); ok {
+				return processDeltas(recorder, store, deltas, isInInitialList)
+			}
+			return errors.New("object given as Process argument is not Deltas")
+		},
+		ProcessBatch: func(deltaList []Delta, isInInitialList bool) error {
+			return processDeltasInBatch(recorder, store, deltaList, isInInitialList)
+		},
+	}
+
+	c := New(cfg)
+	go c.RunWithContext(ctx)
+	if !WaitForCacheSync(ctx.Done(), c.HasSynced) {
+		t.Fatal("Timed out waiting for cache sync")
+	}
+	testReplaceEvents(t, ctx, fifo, recorder, store)
+}
+
+func testReplaceEvents(t *testing.T, ctx context.Context, fifo Queue, m *eventRecorder, store Store) {
+	tcs := []struct {
+		name            string
+		initialObjs     []metav1.Object
+		replacedItems   []interface{}
+		expectedHistory []eventRecord
+	}{
+		{
+			name: "Create with no initial objects",
+			replacedItems: []interface{}{
+				&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod-1", Namespace: "default", ResourceVersion: "1"}},
+			},
+			expectedHistory: []eventRecord{
+				{Action: "add", Key: "default/pod-1", EventRV: "1", StoreRV: "1"},
+			},
+		},
+		{
+			name: "Delete all objects",
+			initialObjs: []metav1.Object{
+				&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod-1", Namespace: "default", ResourceVersion: "1"}},
+				&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod-3", Namespace: "default", ResourceVersion: "1"}},
```

Comment:
(global comment) make sure every revision of every object in a testcase has a unique RV

Reply:
Yep, just thought about that as well, already updated :)